Doc/reference/compound_stmts.rst

```diff
@@ -308,9 +308,10 @@ from a function that handled an exception.
    statement: break
    statement: continue
 
-The optional :keyword:`else` clause is executed if and when control flows off
-the end of the :keyword:`try` clause. [#]_ Exceptions in the :keyword:`else`
-clause are not handled by the preceding :keyword:`except` clauses.
+The optional :keyword:`else` clause is executed if no exception was raised, and
```

Comment:
I don't know how important is to still mention that the control flow has to exit the try block. I can imagine a `yield` statement that interrupts the flow in the middle of the `try` block and never enters again. In this case no `return`,  `continue` or `break` os executed and no exception is raised but the `else` is still not executed because the control flow did not reach outside the `try`.

Reply:
Something like:

> The optional :keyword:`else` clause is executed if **the program execution leaves the "try" block and** no exception was raised, and

Reply:
Hmmm, while I do see your point now, taking yield into account would also suggest modifications to the for and while loops, right? Since their else clauses would also be executed if the control flow leaves the for/while clauses. Perhaps we are getting into too much detail?

Reply:
I don't suggest to even mention "yield" or all the ways the flow can be interrupted, just maintain that the flow has to exit the try block, which is less than a sentence.

This is just a suggestion, thought.

Reply:
I don't suggest to even mention "yield" or all the ways the flow can be interrupted, just maintain that the flow has to exit the try block, which is less than a sentence.

This is just a suggestion, thought.